library.js

```diff
@@ -616,39 +606,43 @@ const fetchUserInfo = async (url, token) => {
   }
 };
 
-const fetchOrganizationInfo = async (url, token) => {
-  const organizationalInfo = await fetchUserInfo(url, token);
-  if (!organizationalInfo) return null;
-
-  const primarySchoolOrganization = organizationalInfo.find(
-    (org) => org.membership?.primarySchool === true,
+const getFeideUser = async (token, feiderUserUrl, validRoles) => {
+  const userInfo = await fetchUserInfo(
+    feiderUserUrl,
+    token,
+    'feideauthorization',
   );
-  const primaryOrFirst = primarySchoolOrganization
-    ? primarySchoolOrganization
-    : organizationalInfo[0];
-  return primaryOrFirst.displayName;
-};
-
-const validateToken = async (token, userInfoUrl) => {
-  const userInfo = await fetchUserInfo(userInfoUrl, token);
-  if (userInfo) {
-    winston.info('ID is valid:', userInfo);
-    return userInfo;
-  }
-  return null;
-};
-
-const validateMemberStatus = async (token, memberStatusUrl, validRoles) => {
-  const userInfo = await fetchUserInfo(memberStatusUrl, token);
   if (
     userInfo &&
-    validRoles.some((role) => userInfo.eduPersonAffiliation.includes(role))
+    validRoles.some((role) => userInfo.role.includes(role)) &&
+    userInfo.arenaEnabled === false
   ) {
-    winston.info('ID is valid for role:', userInfo);
-    return { isValidMember: true, uid: userInfo.uid };
+    const transformedUserInfo = await extractUserInfo(userInfo);
+    return {
+      isValidMember: true,
+      uid: transformedUserInfo.username,
+      userInfo: transformedUserInfo,
+    };
   }
-  winston.warn('[feide-authentication] ID is invalid for role');
   return { isValidMember: false };
 };
 
+const extractUserInfo = async (jsonData) => {
+  const primarySchoolGroup = jsonData.groups.find(
+    (group) => group.isPrimarySchool,
+  );
+
+  return {
+    fullname: jsonData.displayName,
+    sub: jsonData.feideId,
+    email: jsonData.username,
```

Comment:
Ref siste endringa i backend.
```suggestion
    email: jsonData.email,
```